build.go

```diff
@@ -360,3 +364,20 @@ func setStatusDate(build *Build, statusID BuildStatus) {
 		build.CompletedOn.SetValid(now)
 	}
 }
+
+func populateTestResultListSummary(db *gorm.DB, buildID uint, listSummaryPtr *TestResultListSummary) error {
+	listSummaryPtr.BuildID = buildID
+	if err := db.
+		Model(&TestResultSummary{}).
+		Select("sum(failed) as Failed, sum(passed) as Passed, sum(skipped) as Skipped").
+		Where(listSummaryPtr).
+		Scan(listSummaryPtr).
+		Error; err != nil {
+		return err
+	}
+	listSummaryPtr.Total =
+		listSummaryPtr.Failed +
+			listSummaryPtr.Passed +
+			listSummaryPtr.Skipped
+	return nil
+}
```

Comment:
Would rather see this function use return value instead of mutating the argument.

```go
summary, err := getTestResultListSummary(m.Database, buildID)
if err != nil {
	return Build{}, err
}
build.TestResultListSummary = summary

// ...

func getTestResultListSummary(db *gorm.DB, buildID uint, TestResultListSummary) error { }
```

It's a personal preference to prefer "pure functions" (non-mutating functions) as I've had better experience with it and felt that it produces less bugs. Mutating the argument is easily unseen when reading the code, except in some cases such as in `Marshal` and `Unmarshal` functions